services/manifest-repo-export-service/pkg/repository/repository.go

```diff
@@ -812,14 +812,59 @@ func (r *repository) afterTransform(ctx context.Context, transaction *sql.Tx, st
 	return nil
 }
 
-func (r *repository) updateArgoCdApps(ctx context.Context, transaction *sql.Tx, state *State, env string, config config.EnvironmentConfig) error {
+func isAAEnv(config config.EnvironmentConfig) bool {
+	return config.ArgoCdConfigs != nil && len(config.ArgoCdConfigs.ArgoCdConfigurations) > 1
+}
+
+func (r *repository) updateArgoCdApps(ctx context.Context, transaction *sql.Tx, state *State, env string, cfg config.EnvironmentConfig) error {
 	span, ctx := tracer.StartSpanFromContext(ctx, "updateArgoCdApps")
 	defer span.Finish()
 	if !r.config.ArgoCdGenerateFiles {
 		return nil
 	}
+	if isAAEnv(cfg) {
+		for _, currentArgoCdConfiguration := range cfg.ArgoCdConfigs.ArgoCdConfigurations {
+			environmentInfo := &argocd.EnvironmentInfo{
+				ArgoCDConfig:          currentArgoCdConfiguration,
+				CommonPrefix:          *cfg.ArgoCdConfigs.CommonEnvPrefix,
+				ParentEnvironmentName: env,
+				IsAAEnv:               true,
+			}
+			err := r.processArgoAppForEnv(ctx, transaction, state, environmentInfo)
+			if err != nil {
+				return err
+			}
+		}
```

Comment:
nitpick: this could be extracted into a smaller function to avoid code duplication